spec/mongo/socket_spec.rb

```diff
@@ -15,12 +15,19 @@
       end.to raise_error(Mongo::Error::SocketTimeoutError)
     end
 
-    it 'maps SystemCallError and preserves message' do
-      expect do
-        socket.send(:handle_errors) do
-          raise SystemCallError.new('Test error', Errno::ENOMEDIUM::Errno)
-        end
-      end.to raise_error(Mongo::Error::SocketError, 'Errno::ENOMEDIUM: No medium found - Test error')
+    context('only MRI') do
+
+      # On jruby, NameError: uninitialized constant Errno::ENOMEDIUM
+      # https://github.com/jruby/jruby/issues/5257
+      only_mri
```

Comment:
I believe this test would work on jruby. ENOMEDIUM is an arbitrary constant I picked, it doesn't matter which constant it is for the purposes of the test. Can you find an Errno that jruby does implement to use instead of ENOMEDIUM on all platforms?

Reply:
@p-mongo 

> Can you find an Errno that jruby does implement to use instead of ENOMEDIUM on all platforms?

`Errno::ENOMEDIUM::Errno` is 123 on Ruby 2.5.1.
So, that is 123 on jruby?

I checked below things on my local environment Fedora 28.

On jruby-9.1.16.0 on RVM.

```
$ ~/.rvm/rubies/jruby-9.1.16.0/bin/jirb
jruby-9.1.16.0 :001 > Errno::ENOMEDIUM
NameError: uninitialized constant Errno::ENOMEDIUM
Did you mean?  Errno::ENOMEM
	from org/jruby/RubyModule.java:3372:in `const_missing'
	from (irb):1:in `<eval>'
	from org/jruby/RubyKernel.java:995:in `eval'
	from org/jruby/RubyKernel.java:1316:in `loop'
	from org/jruby/RubyKernel.java:1138:in `catch'
	from org/jruby/RubyKernel.java:1138:in `catch'
	from /home/jaruga/.rvm/rubies/jruby-9.1.16.0/bin/jirb:13:in `<main>'

jruby-9.1.16.0 :002 > Errno::ETIMEDOUT
 => Errno::ETIMEDOUT 
```

On Ruby 2.5.1

```
$ irb
irb(main):001:0> Errno::ENOMEDIUM
=> Errno::ENOMEDIUM
irb(main):004:0> Errno::ENOMEDIUM::Errno
=> 123
irb(main):002:0> Errno::ETIMEDOUT
=> Errno::ETIMEDOUT
```

Reply:
According to https://ruby-doc.org/core-2.5.1/Errno.html
> Module Errno is created dynamically

I compared `Errno.constants`. Some values are not defined on the JRuby.
Ref: https://github.com/jruby/jruby/issues/5257#issuecomment-407046281

Reply:
OK, just pick one that's present in both (but not timeout as we handle that specially).

Reply:
OK, I picked this `Errno::ENFILE` that is present in both.

```
irb(main):005:0> raise Errno::ENFILE
Traceback (most recent call last):
        2: from /usr/local/ruby-2.5.1/bin/irb:11:in `<main>'
        1: from (irb):5
Errno::ENFILE (Too many open files in system)
```